OptimoveSDK/Sources/Classes/PreferenceCenter/PreferenceCenterModels.swift

```diff
@@ -0,0 +1,91 @@
+import Foundation
+
+struct PreferenceCenterConfig: Codable {
+    var region: Region
+    var tenantId: Int
+    var brandGroupId: String
+}
+
+public class Preferences: NSObject, Codable {
+    var topics: [Topic]
+    var channels: [Channel]
+
+    public func getChannels() -> [Channel] {
+        return channels
+    }
+
+    public func getTopics() -> [Topic] {
+        return topics
+    }
+
+    public init(topics: [Topic], channels: [Channel]) {
+        self.topics = topics
+        self.channels = channels
+    }
+
+    required public init(from decoder: any Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.topics = try container.decode([Topic].self, forKey: .topics)
+        self.channels = try container.decode([Channel].self, forKey: .channels)
+    }
+}
+
+public enum Channel: Int, Codable {
+    case mobilePush = 489
+    case webPush = 490
+    case sms = 493
+}
+
+public class Topic: NSObject, Codable {
+    public var topicId: String
+    var topicName: String
+    var topicDescription: String
+    var channelSubscription: [Channel]
+
+    public func getId() -> String {
+        return topicId
+    }
```

Comment:
Instead of writing getters, Swift Lang provides private(set) annotation for public fields